lib/fakefs/file.rb

```diff
@@ -279,16 +279,8 @@ def self.executable?(filename)
 
     def self.chown(owner_int, group_int, filename)
       file = FileSystem.find(filename)
-      if owner_int && owner_int != -1
-        owner_int.is_a?(Fixnum) || fail(TypeError,
-                                        "can't convert String into Integer")
-        file.uid = owner_int
-      end
-      if group_int && group_int != -1
-        group_int.is_a?(Fixnum) || fail(TypeError,
-                                        "can't convert String into Integer")
-        file.gid = group_int
-      end
+      file.uid = Integer(owner_int, 10) if owner_int && owner_int != -1
```

Comment:
actually this is wrong ... we cannot convert string to integer without changing behavior ... and behavior should be based on original chown ... so either make sure it works the same in the original or use `.is_a?(Integer)`